docs/src/snap/howto/2-node-ha.md

```diff
@@ -0,0 +1,423 @@
+
+sudo drbdadm create-md r0
+sudo drbdadm status
+```
+
+Let's create a mount point for the DRBD block device. Non-default mount points
+need to be passed to the 2ha.sh script mentioned above, see the script for the
+full list of configurable parameters.
+
+```
+DRBD_MOUNT_DIR=/mnt/drbd0
+sudo mkdir -p $DRBD_MOUNT_DIR
+```
+
+Run the following *once* to initialize the filesystem.
+
+```
+sudo drbdadm up r0
+
+sudo drbdadm -- --overwrite-data-of-peer primary r0/0
+sudo mkfs.ext4 /dev/drbd0
+
+sudo drbdadm down r0
+```
+
+Add the drbd device to the ``multipathd`` blacklist, ensuring that the multipath
+service will not attempt to manage this device:
+
+```
+sudo cat <<EOF | sudo tee -a /etc/multipath.conf
+blacklist {
+    devnode "^drbd*"
+}
+EOF
+
+sudo systemctl restart multipathd
+```
+
+### Corosync
+
+Let's prepare the Corosync configuration. Again, make sure to use the correct
+addresses.
+
+```
+HAONE_ADDR=<firstNodeAddress>
+HATWO_ADDR=<secondNodeAddress>
+
+sudo mv /etc/corosync/corosync.conf /etc/corosync/corosync.conf.orig
+
+cat <<EOF | sudo tee /etc/corosync/corosync.conf
+totem {
+  version: 2
+  cluster_name: ha
+  secauth: off
+  transport:udpu
+  interface {
+    ringnumber: 0
+    bindnetaddr: ${netaddr}
+    broadcast: yes
+    mcastport: 5405
+  }
+}
+
+nodelist {
+  node {
+    ring0_addr: ${HAONE_ADDR}
+    name: haone
+    nodeid: 1
+  }
+  node {
+    ring0_addr: ${HATWO_ADDR}
+    name: hatwo
+    nodeid: 2
+  }
+}
+
+quorum {
+  provider: corosync_votequorum
+  two_node: 1
+  wait_for_all: 1
+  last_man_standing: 1
+  auto_tie_breaker: 0
+}
+EOF
+```
+
+Follow the above steps on both nodes before moving forward.
+
+### Pacemaker
+
+Let's define a Pacemaker resource for the DRBD block device, which
+ensures that the block device will be mounted on the replica in case of a
+primary node failure.
+
+Pacemaker fencing (stonith) configuration is environment specific and thus
+outside the scope of this guide. However, we highly recommend using fencing
+if possible to reduce the risk of cluster split-brain situations.
+
+```
+HAONE_ADDR=<firstNodeAddress>
+HATWO_ADDR=<secondNodeAddress>
+DRBD_MOUNT_DIR=${DRBD_MOUNT_DIR:-"/mnt/drbd0"}
+
+sudo crm configure <<EOF
+property stonith-enabled=false
+property no-quorum-policy=ignore
+primitive drbd_res ocf:linbit:drbd params drbd_resource=r0 op monitor interval=29s role=Master op monitor interval=31s role=Slave
+ms drbd_master_slave drbd_res meta master-max=1 master-node-max=1 clone-max=2 clone-node-max=1 notify=true
+primitive fs_res ocf:heartbeat:Filesystem params device=/dev/drbd0 directory=${DRBD_MOUNT_DIR} fstype=ext4
+colocation fs_drbd_colo INFINITY: fs_res drbd_master_slave:Master
+order fs_after_drbd mandatory: drbd_master_slave:promote fs_res:start
+commit
+show
+quit
+EOF
+```
+
+Before moving forward, let's ensure that the DRBD Pacemaker resource runs on
+the primary (voter) Dqlite node.
+
+Remember that in our case only the primary node contains the latest Dqlite data,
```

Comment:
In this setup, only the primary node holds the latest Dqlite data, which will be transferred to the DRBD device